base_request.py

```diff
@@ -0,0 +1,149 @@
+import requests
+import json
+import resin
+import urllib
+
+from urlparse import urljoin
+
+from .settings import Settings
+from .token  import Token
+from .resources import Message
+
+from exceptions import (
+    JSONDecodeError,
+    NoTokenProvided,
+    ResponseError,
+    RequestError
+)
+
+
+class BaseRequest(object):
+
+    def __init__(self):
+        self.settings = Settings()
+        self.token = Token()
+        self.util = Util()
+
+    def __str__(self):
+        return b'<{:s} at {:#x}>'.format(type(self).__name__, id(self))
+
+    def __unicode__(self):
+        return '<{:s} at {:#x}>'.format(type(self).__name__, id(self))
```

Comment:
@dfunckt , this is from the old sources, I've not touched this. Anw, removed because It's not used anywhere